ffi/athcon/include/athcon/athcon.h

```diff
@@ -563,6 +577,9 @@ extern "C"
 
     /** Get block hash callback function. */
     athcon_get_block_hash_fn get_block_hash;
+
+    /** Spawn program callback function. */
+    athcon_spawn_fn spawn;
```

Comment:
The Go bindings need to be updated too. The build emits the following warning:
```
# github.com/athenavm/athena/ffi/athcon/bindings/go [github.com/athenavm/athena/ffi/athcon/bindings/go.test]
host.c:19:1: warning: missing initializer for field ‘spawn’ of ‘const struct athcon_host_interface’ [-Wmissing-field-initializers]
   19 | };
      | ^
In file included from host.go:6,
                 from host.c:1:
./../../include/athcon/athcon.h:5[82](https://github.com/athenavm/athena/actions/runs/10428366033/job/28884425903?pr=74#step:4:83):21: note: ‘spawn’ declared here
  582 |     athcon_spawn_fn spawn;
      |                     ^~~~~
```

Perhaps we should add `-Werror` to CFLAGS, wdyt?